modules/otel-collector-components.adoc

```diff
@@ -869,7 +902,47 @@ This extension supports traces, metrics, and logs.
 <10> The path to the client's private key file that is used with the client certificate if needed for authentication.
 <11> Sets a timeout for the token client's request.
 <12> You can assign the authenticator configuration to an OTLP exporter.
+[id="filestorage-extension_{context}"]
+=== Filestorage extension
+
+The File Storage extension can persist state to the local file system.
+The extension requires read and write access to a directory. A default directory can be used, but it must already exist in order for the extension to operate.
+
+The OAuth2Client extension persists the sending queue for the OTLP exporters that are based on the HTTP and the gRPC protocol.
+This extension supports traces, metrics, and logs.
 
+.OpenTelemetry Collector custom resource with configured FileStorage extension to persist as OTLP sendig queue.
```

Comment:
@frzifus, could you please reply in this thread to explain this phrase to me the way you understand it: `to persist as OTLP sendig queue` ?

Reply:
The OTLP exporter has a sending queue where all messages that should be published get "parked" until they are successfully transmitted.
By default this is in memory. If you restart the collector, this data will be lost.

The file storage extension implements the interface of this queue and writes the messages to disk. If you restart the collector, all non transmitted messages should still exist.